crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,268 @@
+        let right = binary_expression.right().ok()?;
+        let shorhand = get_shorthand(&operator)?;
+
+        if has_same_reference {
+            Some(RuleState {
+                shorthand_operator: shorhand,
+                right,
+            })
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let shorthand_operator = state.shorthand_operator.to_string()?;
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Assignment (=) can be replaced with operator assignment "{shorthand_operator}""
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let mut mutation = ctx.root().begin();
+        let shorthand_operator = state.shorthand_operator;
+
+        let token_leading_trivia = node.operator_token().ok()?.leading_trivia().pieces();
+        let token_trailing_trivia = node.operator_token().ok()?.trailing_trivia().pieces();
+        let token = make::token(shorthand_operator)
+            .with_leading_trivia_pieces(token_leading_trivia)
+            .with_trailing_trivia_pieces(token_trailing_trivia);
+
+        let next_node = node
+            .clone()
+            .with_operator_token_token(token)
+            .with_right(state.right.clone());
+
+        mutation.replace_node(
+            AnyJsExpression::JsAssignmentExpression(node.clone()),
+            AnyJsExpression::JsAssignmentExpression(next_node),
+        );
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! { "Use "<Emphasis>""{shorthand_operator.to_string()?}""</Emphasis>" instead." }
+                .to_owned(),
+            mutation,
+        })
+    }
+}
+
+fn has_same_reference_in_expression(
+    variable_name: String,
+    binary_expression: &JsBinaryExpression,
+) -> Option<bool> {
+    Some(variable_name == binary_expression.left().ok()?.omit_parentheses().text())
+}
+
+fn get_shorthand(operator: &JsBinaryOperator) -> Option<JsSyntaxKind> {
+    match operator {
+        JsBinaryOperator::Plus => Some(T![+=]),
+        JsBinaryOperator::Minus => Some(T![-=]),
+        JsBinaryOperator::Times => Some(T![*=]),
+        JsBinaryOperator::Divide => Some(T![/=]),
+        JsBinaryOperator::Remainder => Some(T![%=]),
+        JsBinaryOperator::Exponent => Some(T![**=]),
+        JsBinaryOperator::BitwiseAnd => Some(T![&=]),
+        JsBinaryOperator::BitwiseOr => Some(T![|=]),
+        JsBinaryOperator::BitwiseXor => Some(T![^=]),
+        JsBinaryOperator::LeftShift => Some(T![<<=]),
+        JsBinaryOperator::RightShift => Some(T![>>=]),
+        JsBinaryOperator::UnsignedRightShift => Some(T![>>>=]),
+        _ => None,
```

Comment:
I believe this method is not needed. I think you can do `operator.token().ok()?.kind()`

Reply:
@ematipico That's much better, i've only added this function to filter some operators that doesn't has shorthand:

    LessThan,
    GreaterThan,
    LessThanOrEqual,
    GreaterThanOrEqual,
    Equality,
    StrictEquality,
    Inequality,
    StrictInequality

Reply:
Maybe i can do different,
Instead use `get_shorthand` i can check and return early for not valid operators

Reply:
Oh, maybe `.token()` should be enough?

Reply:
We can leave it as it is for now, and come back later. Without playing with the code, I can't give you a proper suggestion